compiler/optimizer/DeadTreesElimination.cpp

```diff
@@ -339,7 +346,20 @@ static bool isSafeToReplaceNode(TR::Node *currentNode, TR::TreeTop *curTreeTop,
       int32_t tempHeight = 0;
       int32_t maxHeight = 0;
       bool canMoveIfVolatile = true;
-      if (containsNode(node, currentNode, visitCount, comp, &tempHeight, &maxHeight, canMoveIfVolatile))
+      bool nodeInSubTree = containsNode(node, currentNode, visitCount, comp, &tempHeight, &maxHeight, canMoveIfVolatile);
```

Comment:
This really should have some trace to help debugging IMO

Reply:
Added trace at the place deciding to swing down volatile load. inside `if (nodeInSubTree)` and before returns true.